src/contract/aci.ts

```diff
@@ -0,0 +1,487 @@
+/*
+* ISC License (ISC)
+* Copyright (c) 2022 aeternity developers
+*
+*  Permission to use, copy, modify, and/or distribute this software for any
+*  purpose with or without fee is hereby granted, provided that the above
+*  copyright notice and this permission notice appear in all copies.
+*
+*  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
+*  REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+*  AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+*  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+*  LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
+*  OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+*  PERFORMANCE OF THIS SOFTWARE.
+*/
+
+/**
+ * ContractACI module
+ *
+ * @module @aeternity/aepp-sdk/es/contract/aci
+ * @export getContractInstance
+ */
+
+// @ts-expect-error TODO remove
+import { Encoder as Calldata } from '@aeternity/aepp-calldata'
+// @ts-expect-error TODO remove
+import { DRY_RUN_ACCOUNT, GAS_MAX } from '../tx/builder/schema'
+// @ts-expect-error TODO remove
+import TxObject from '../tx/tx-object'
+// @ts-expect-error TODO remove
+import { decode } from '../tx/builder/helpers'
+import {
+  MissingContractDefError,
+  MissingContractAddressError,
+  InactiveContractError,
+  BytecodeMismatchError,
+  DuplicateContractError,
+  MissingFunctionNameError,
+  InvalidMethodInvocationError,
+  NotPayableFunctionError,
+  TypeError,
+  NodeInvocationError,
+  IllegalArgumentError,
+  NoSuchContractFunctionError,
+  MissingEventDefinitionError,
+  AmbiguousEventDefinitionError
+} from '../utils/errors'
+import { hash } from '../utils/crypto'
+
+interface FunctionACI {
+  arguments: any[]
+  name: string
+  payable: boolean
+  returns: string
+  stateful: boolean
+}
+
+interface Aci {
+  encodedAci: {contract: {
+    name: string
+    event: any
+    kind: string
+    state: any
+    type_defs: any[]
+    functions: FunctionACI[]
+  }}
+  externalEncodedAci: any[]
+  interface: string
+}
```

Comment:
I propose to enchance the ACI definition that compiler has in swagger file by adding a directive in https://github.com/aeternity/aepp-sdk-js/blob/777e990f2e900b4d8df8b4c4d91c1d4e9bab6f72/tooling/autorest/compiler.yaml#L17-L16, instead of doing an extra aci defnintion.
Later we will make a PR with it to compiler repo.

Reply:
I haven’t worked with autorest or swagger in the past so I don’t think I’m the right person to make the changes required in the `complier.yaml` file. For now I've just extended my ACI interface with the compiled ACI interface.